File: pre-AEDS2/binary_search.py
list = [0,1,2,3,4,5,6,7,8,9]

def binary_search(list, target):
    start = 0
    end = len(list)-1
    i = 0

    while start <= end :
        mid = (start + end)//2
        guess = list[mid]

        if guess == target:
            return  f"Found {mid} in {i} tries"

        
        if guess > target:
            end = mid - 1
            i += 1
            

        else:
            start = mid +1
            i += 1
            

    return None



def main():
    target = int(input("Insira valor a procurar: "))
    print(binary_search(list, target))

if __name__ == "__main__":
    main()

File: pre-AEDS2/binarySearch.java
import java.util.Scanner;

public class Main {

    public static int binarySearch(int[] list, int target) {
        int start = 0;
        int end = list.length - 1;
        int tries = 0;

        while (start <= end) {
            tries++;
            int mid = (start + end) / 2;
            int guess = list[mid];

            if (guess == target) {
                System.out.println("Found at index " + mid + " in " + tries + " tries");
                return mid;
            } 
            else if (guess > target) {
                end = mid - 1;
            } 
            else {
                start = mid + 1;
            }
        }

        System.out.println("Not found after " + tries + " tries");
        return -1;
    }

    public static void main(String[] args) {
        int[] list = {1,2,3,4,5,6,7,8,9};

        Scanner scanner = new Scanner(System.in);
        int target = scanner.nextInt();

        binarySearch(list, target);
        scanner.close();
    }
}
